src/main/java/Bob.java

```diff
@@ -0,0 +1,430 @@
+                if (!directoriesCreated) {
+                    System.out.println("Failed to create directories.");
+                    return;
+                }
+                System.out.println("Directories created: " + data.getAbsolutePath());
+            }
+
+            if (!bob.exists()) {
+                boolean fileCreated = bob.createNewFile();
+                if (!fileCreated) {
+                    System.out.println("Failed to create file.");
+                    return;
+                }
+                System.out.println("File created: " + bob.getAbsolutePath());
+            }
+
+            FileWriter writer = new FileWriter(bob);
+            BufferedWriter bufferedWriter = new BufferedWriter(writer);
+
+            for (Task task : taskList) {
+                String taskType;
+                String line;
+                String isDone = task.getIsDone() ? "1" : "0";
+                String description = task.getDescription();
+
+                if (task instanceof Deadline) {
+                    taskType = "D";
+                    LocalDateTime deadlineDate = ((Deadline) task).getBy();
+                    String deadline = deadlineDate.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm"));
+                    line = String.format("%s | %s | %s | %s", taskType, isDone, description, deadline);
+                }
+
+                else if (task instanceof Event) {
+                    taskType = "E";
+                    String from = ((Event) task).getFrom();
+                    String to = ((Event) task).getTo();
+                    line = String.format("%s | %s | %s | %s | %s", taskType, isDone, description, from, to);
+                }
+
+                else {
+                    taskType = "T";
+                    line = String.format("%s | %s | %s", taskType, isDone, description);
+                }
+
+                bufferedWriter.write(line);
+                bufferedWriter.newLine();
+            }
+            bufferedWriter.close();
+            System.out.println("File saved.");
+        }
+
+        catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+}
+
+/*
+ * This class represents a task we want to record.
+ */
+class Task {
+    private String description;
+    private boolean isDone;
+
+    /*
+     * A constructor that depicts a new task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /*
+     * A method that marks a task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /*
+     * A method that will undo the mark on a task.
+     */
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+
+    /*
+     * A method that returns the isDone boolean.
+     *
+     * @return A boolean depending on whether the task is done.
+     */
+    public boolean getIsDone() {
```

Comment:
By java convention, methods that start with get() usually return a Value like a string, integer or array. For booleans you can just use isDone()